Bytesize transcript ampliseq (#1532)
nf-core/nf-co.re: markdown/events/2021/bytesize-25-nf-core-ampliseq.md

[7:50](https://youtu.be/a0VOEeAvETs&t=470)
I did that not only with one sample, but with three different samples and with here the OTU-based methods and the ASV-based methods, not only for precision and sensitivity of finding the original sample sequences again, but also for the taxonomic classification and for a diversity index, the Shannon diversity. Here the green color shows the best methods and the red ones, the worst ones. You can see that the ASV-based methods perform much better in the amplicon sequencing analyses than the OTU-based methods.

[8:31](https://youtu.be/a0VOEeAvETs&t=511)

> Coming from that, I wanted to produce a pipeline that is doing exactly what I need and to use that kind of knowledge that I have accumulated by the benchmarking. I wanted to produce a pipeline to analyze all kinds of amplicon data with a wide range of input types, reference taxonomies, and downstream analyses. We came up in 2018 then with the nf-core/ampliseq pipeline, in the end of 2018. We released version 1.00 and did some feature and maintenance updates. We also then published a paper about it and continued afterwards in 2021 quite a lot with adding more features to the pipeline. Now it is at a pretty good stage how I imagined it in the beginning.

Reviewer: `version 1.0.0` instead of `version 1.00`
```suggestion
Coming from that, I wanted to produce a pipeline that is doing exactly what I need and to use that kind of knowledge that I have accumulated by the benchmarking. I wanted to produce a pipeline to analyze all kinds of amplicon data with a wide range of input types, reference taxonomies, and downstream analyses. We came up in 2018 then with the nf-core/ampliseq pipeline, in the end of 2018. We released version 1.0.0 and did some feature and maintenance updates. We also then published a paper about it and continued afterwards in 2021 quite a lot with adding more features to the pipeline. Now it is at a pretty good stage how I imagined it in the beginning.
```

[10:57](https://youtu.be/a0VOEeAvETs&t=657)
Depending on what kind of sequencing data from what kind of sequencing technology you have, you have to also add additional parameters. For Illumina paired-end sequencing, just default, you don't need to add anything else. For single-end, the `--single_end` parameter, for PacBio and for IonTorrent technologies, you have to specify that. In special cases, Illumina paired-end ITS, so fungi community analyses, you also should already set a parameter so that all the downstream analyses will have sensible settings.

[11:38](https://youtu.be/a0VOEeAvETs&t=698)

> What is the output of those very basic analyses? In the results folder, in our subfolder "data" will be that file, and that file here contains already most of the information that you will need. It contains for each of the different samples, the sequences that it found and also the quantification, so it simply counts how many of the sequences in there will originate from that original 16S sequence in your sample. There is also a handy overall summary for all the read numbers that were for each of the samples processed and then ended up in the table and if those numbers do not look as expected, then this is a very good starting point to troubleshoot.

Reviewer: Its actually "dada2", I forgot the 2 I guess
```suggestion
What is the output of those very basic analyses? In the results folder, in our subfolder "dada[2]" will be that file, and that file here contains already most of the information that you will need. It contains for each of the different samples, the sequences that it found and also the quantification, so it simply counts how many of the sequences in there will originate from that original 16S sequence in your sample. There is also a handy overall summary for all the read numbers that were for each of the samples processed and then ended up in the table and if those numbers do not look as expected, then this is a very good starting point to troubleshoot.
```

[12:34](https://youtu.be/a0VOEeAvETs&t=754)

> The next step of the pipeline is to classify the sequences that we have now produced, meaning that we give it a name, that we give it a taxonomic name. This is by default done with the DADA2 software, but we also have an alternative way to do this with QIIME2 which is also a very popular program in the area. By default it is using the DADA algorithm with the SILVA138 taxonomy. So we have here a range of reference taxonomy databases that can be used and that is the SILVA, the RDP, the GTDB database that are all for bacteria, for 16S rRNA gene amplicon analyses; the UNITE database for her fungi analyzes ITS, internal transcribed spacers, and the PR2 database for eukaryotes with the 18S rRNA gene amplicons.

Reviewer: ```suggestion
The next step of the pipeline is to classify the sequences that we have now produced, meaning that we give it a name, that we give it a taxonomic name. This is by default done with the DADA2 software, but we also have an alternative way to do this with QIIME2 which is also a very popular program in the area. By default it is using the DADA algorithm with the SILVA138 taxonomy. So we have here a range of reference taxonomy databases that can be used and that is the SILVA, the RDP, the GTDB database that are all for bacteria, for 16S rRNA gene amplicon analyses; the UNITE database for fungi analyzes ITS, internal transcribed spacers, and the PR2 database for eukaryotes with the 18S rRNA gene amplicons.
```

[14:37](https://youtu.be/a0VOEeAvETs&t=877)
Finally there is also downstream analysis, but this downstream analysis is only provided when the optional parameter metadata is pointing to a tab-separated metadata sheet. Then automatically appropriate metadata columns will be chosen and used for visualization of a bar plot, as you can see here. Those are HTML files that are then interactive so you can choose colors, you can choose different kinds of taxonomic levels. I chose here a very high one just to show you, for example here with the test data set, that we are typically running here with triplicates, that you can see already patterns of different kind of data.

[15:27](https://youtu.be/a0VOEeAvETs&t=927)

> Additionally there will be a differential abundance analysis with a program called ANCOM and that produces for example such a volcano plot, also interactive. This red dot here which is identified as a significant different abundance between the different treatments would be a burkholderiales bacteria.

Reviewer: Taxonomic names are typically capitalized, often in italics
```suggestion
Additionally there will be a differential abundance analysis with a program called ANCOM and that produces for example such a volcano plot, also interactive. This red dot here which is identified as a significant different abundance between the different treatments would be a _Burkholderiales_ bacteria.
```

[15:52](https://youtu.be/a0VOEeAvETs&t=952)
Then also alpha and beta diversity indices are produced. Alpha diversity is a measure of each of the samples, how diverse the sequences in there are. You can see that for this test data set, it's a bit small, the samples originating from groundwater, have the lowest Shannon diversity index but the sediment samples here have the highest one. To all of these comparisons will be of course also statistical analyses provided in the output of the pipeline. Beta diversity plots show the difference in PCOA plots. What you can see here is that the replicates for the different kinds of sample sources (groundwater, river water, sediment and soil in our example data set) are nicely clustering and well separated from the others.

[16:55](https://youtu.be/a0VOEeAvETs&t=1015)

> Additionally there will be quality control figures like alpha rarefaction curves, which show you if you have a sequence with sufficient output for your samples. As long as those curves will flatten out you know that you have sufficient sequencing data produced for that sample. All those results can be also found on the nf-core website in `/ampliseq/results` and you can have a look what is produced, how does it look like and there is also extensive documentation what each of those plots mean and what you can use it for. Those plots, of course, cannot show all kinds of complicated experimental setups, but they are very helpful for getting a first view on the data and are also a sort of quality control that what you are producing here makes sense. With this I want to finish and I want to thank you for your attention, my colleagues at QBiC , who also were involved in the development of the pipeline, nf-core, especially Daniel Lundin, helped a lot and the University of Tübingen Microbiology and Geomicrobiology who produced a lot of data, that I was then able to analyze and with this pipeline and produce it as good as possible.

Reviewer: Not sure, will links work? see [`/ampliseq/results`](https://nf-co.re/ampliseq/results)
```suggestion
Additionally there will be quality control figures like alpha rarefaction curves, which show you if you have a sequence with sufficient output for your samples. As long as those curves will flatten out you know that you have sufficient sequencing data produced for that sample. All those results can be also found on the nf-core website in [`/ampliseq/results`](https://nf-co.re/ampliseq/results) and you can have a look what is produced, how does it look like and there is also extensive documentation what each of those plots mean and what you can use it for. Those plots, of course, cannot show all kinds of complicated experimental setups, but they are very helpful for getting a first view on the data and are also a sort of quality control that what you are producing here makes sense. With this I want to finish and I want to thank you for your attention, my colleagues at QBiC , who also were involved in the development of the pipeline, nf-core, especially Daniel Lundin, helped a lot and the University of Tübingen Microbiology and Geomicrobiology who produced a lot of data, that I was then able to analyze and with this pipeline and produce it as good as possible.
```

Author: they should. If there is an issue, I can solve it later separately

[18:33](https://youtu.be/a0VOEeAvETs&t=1113)
(host) Thanks very much for that very clear and insightful talk Daniel. Are there any questions? If anyone in the audience has questions you can unmute yourself or even add questions to the chat and I can read them out. Okay I don't see anything coming up so that was extremely clear so I guess that's why people don't have questions but if anything crops up anywhere, you know where to reach Daniel. You can message him and continue the discussion on the bytesize channel and thanks again for joining. We will be back next week with another pipeline focused talk.

[19:23](https://youtu.be/a0VOEeAvETs&t=1163)
(question) We have one comment here okay there's a question. This person thanks you for the presentation and asks what type of license is associated with this kind of tool, for example the ampliseq pipeline.

> (answer) It is a CCPY license, like I think all the nf-core pipelines have, there are no restrictions whatsoever.

Reviewer: I know I made a mistake here, it is not CCBY, that is only the overview figure, is it possible to note that this is wrong and its MIT license (default for nf-core pipelines I think)?

Author: I can add a comment
